metadata-ingestion/tests/unit/test_redshift_lineage.py

```diff
@@ -149,3 +183,42 @@ def test_cll():
             logic=None,
         ),
     ]
+
+
+def test_collapse_temp_lineage():
+    lineage_extractor = get_lineage_extractor()
+
+    lineage_extractor.get_temp_tables = lambda connection: [  # type: ignore
+        TempTableRow(
+            transaction_id=126,
+            query_text="CREATE TABLE #player_price distkey(player_id) AS SELECT player_id, SUM(price) AS price_usd "
+            "from player_activity group by player_id",
+            start_time=datetime.now(),
+            session_id="abc",
+            create_command="CREATE TABLE #player_price",
+        ),
+    ]
+
+    datasets: List[LineageDataset] = lineage_extractor._get_upstream_lineages(
+        sources=[
+            LineageDataset(
+                platform=LineageDatasetPlatform.REDSHIFT,
+                urn="urn:li:dataset:(urn:li:dataPlatform:redshift,dev.public.#player_price,PROD)",
+            )
+        ],
+        target_table="urn:li:dataset:(urn:li:dataPlatform:redshift,dev.public.player_price_with_hike_v4,PROD)",
+        raw_db_name="dev",
+        alias_db_name="dev",
+        all_tables_set={
+            "dev": {
+                "public": set(),
+            }
+        },
+        connection=MagicMock(),
+    )
+
+    assert len(datasets) == 1
+    assert (
+        datasets[0].urn
+        == "urn:li:dataset:(urn:li:dataPlatform:redshift,dev.public.player_activity,PROD)"
```

Comment:
let's add tests for CLL here

it looks like the recursive merging for CLL is not working